velox/common/file/tests/FileTest.cpp

```diff
@@ -215,3 +215,26 @@ TEST(LocalFile, readFileDestructor) {
     ASSERT_ANY_THROW(readData(&readFile, false));
   }
 }
+
+TEST(LocalFile, mkdir) {
+  filesystems::registerLocalFileSystem();
+  auto tempFolder = ::exec::test::TempDirectoryPath::create();
+
+  std::string path = tempFolder->path;
+  auto localFs = filesystems::getFileSystem(path, nullptr);
+
+  // Create 3 levels of directories and ensure they exist.
+  path += "/level1/level2/level3";
+  EXPECT_TRUE(localFs->mkdir(path));
+  EXPECT_TRUE(localFs->exists(path));
+
+  // Write a file to our directory and ensure it exist.
+  path += "/a.txt";
+  const std::string data("aaaaa");
+  {
+    auto writeFile = localFs->openFileForWrite(path);
+    writeFile->append(data);
+    writeFile->close();
+  }
+  EXPECT_TRUE(localFs->exists(path));
```

Comment:
Please add test cases that parent directory exists (A/B/C A A exists, A/B exists).

Also could you add to test openfile for write will fail if parent directory doesn't exist?

Reply:
> Please add test cases that parent directory exists (A/B/C A A exists, A/B exists).

This is covered by

`auto tempFolder = ::exec::test::TempDirectoryPath::create();`

> Also could you add to test openfile for write will fail if parent directory doesn't exist?

Don't we have a test where openFile fails due to file not existing? How that's different from what you are suggesting?

Reply:
Openfile for write will create the new file. I mean could we catch the behavior that if the parent directory doesn't exist, does the open succeed or not? Might guess different fs might behave differently. It would be nice if we could catch these in test. Thanks!

Reply:
> Openfile for write will create the new file. I mean could we catch the behavior that if the parent directory doesn't exist, does the open succeed or not? Might guess different fs might behave differently. It would be nice if we could catch these in test. Thanks!

I expected us to have the test for the file write, but just checked and we don't have such.
IMHO, write file failure for non-existing directory has little to do with the mkdir test (feature creep).
So, a separate test should be created or a write file test upgraded to contain that.
Once again, that would not belong to this change, strictly speaking.
I can add such test separately.